Fix k_sleep() timeouts in net

jukkar opened the pull request:
This converts remaining MSEC() calls to K_MSEC() in networking code.
Also convert MSEC_PER_SEC to K_SECONDS() as that is more intuitive.

Fixes #7657

jukkar committed:
net: Too long timeout for k_sleep

Convert couple of MSEC() calls to K_MSEC() as the timeouts
when using MSEC() are just too long.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
net: Convert MSEC_PER_SEC to K_SECONDS()

Use of K_SECONDS() macro is more intuitive so use that instead of
plain MSEC_PER_SEC define.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

pfalcon commented on subsys/net/ip/ipv6.c:
```diff
@@ -65,7 +65,7 @@ static void nd_reachable_timeout(struct k_work *work);
 #define MAX_UNICAST_SOLICIT   3
-#define DELAY_FIRST_PROBE_TIME (5 * MSEC_PER_SEC) /* RFC 4861 ch 10 */
+#define DELAY_FIRST_PROBE_TIME K_SECONDS(5) /* RFC 4861 ch 10 */
 #define RETRANS_TIMER 1000 /* in ms, RFC 4861 ch 10 */
```
We have more "raw" values like this, IMHO makes sense to convert them within the scope of this PR too.

pfalcon commented on subsys/net/ip/l2/ieee802154/ieee802154_fragment.c:
```diff
@@ -26,8 +26,8 @@
-#define FRAG_REASSEMBLY_TIMEOUT (MSEC_PER_SEC * \
-				 CONFIG_NET_L2_IEEE802154_REASSEMBLY_TIMEOUT)
+#define FRAG_REASSEMBLY_TIMEOUT \
+	K_SECONDS(CONFIG_NET_L2_IEEE802154_REASSEMBLY_TIMEOUT)
```
Should convert this to msecs for better granularity. In a separate PR apparently.

pfalcon commented on subsys/net/ip/rpl.c:
```diff
@@ -1642,8 +1642,7 @@ static void set_dao_lifetime_timer(struct net_rpl_instance *instance)
 		expiration_time = (u32_t)instance->default_lifetime *
-			(u32_t)instance->lifetime_unit *
-			MSEC_PER_SEC / 2;
+			(u32_t)instance->lifetime_unit * K_SECONDS(1) / 2;
```
Maybe K_MSEC(500) after all?

jukkar replied:
The comment few lines above says "Set up another DAO within half the expiration time" so in this respect it makes sense to leave the division in place.

pfalcon reviewed (commented):
Good! (Was in my queue too ;-) )

Suggested to over "raw" time-related values in net/ip/ and convert them too.

jhedberg commented on subsys/net/buf.c:
```diff
@@ -289,11 +289,11 @@ struct net_buf *net_buf_alloc_len(struct net_buf_pool *pool, size_t size,
 				     func, line, pool->name,
-				     (k_uptime_get_32() - ref) / MSEC_PER_SEC);
+				     (k_uptime_get_32() - ref) / K_SECONDS(1));
 #else
```
The rest of the fixes in this PR seem good, but I'm not sure I agree with this. The way the K_SECONDS, K_MSEC, K_MINUTES, K_HOURS macros work, is that you don't need to know what the "native" representation of time is that's passed to the kernel APIs. It might be milliseconds, but it could be something else that the macros convert the input to. The above change assumes that the macro converts to milliseconds, whereas with MSEC_PER_SEC it's at least explicit.

pfalcon replied:
+1 to @jhedberg's comment, I'd also say that division by MSEC_PER_SEC is easier to understand than division by K_SECONDS(1).

jukkar replied:
Ok, lets leave MSEC_PER_SEC there then.

jhedberg commented on subsys/net/buf.c:
```diff
@@ -289,11 +289,11 @@ struct net_buf *net_buf_alloc_len(struct net_buf_pool *pool, size_t size,
 				     func, line, pool,
-				     (k_uptime_get_32() - ref) / MSEC_PER_SEC);
+				     (k_uptime_get_32() - ref) / K_SECONDS(1));
 #endif
```
Same here.

jukkar committed:
net: Convert raw timeout values to use K_MSEC() macro

This one converts "raw" timeout value to use K_MSEC() macro
in order to make clear how long the timeout is.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar commented:
Updated according to comments.
